alf/algorithms/mcts_algorithm.py

```diff
@@ -317,8 +351,15 @@ def predict_step(self, time_step: TimeStep, state: MCTSState):
                 action_probs=model_output.action_probs *
                 valid_action_mask.to(torch.float32))
 
-        trees = _MCTSTree(self._num_simulations + 1, model_output,
-                          self._known_value_bounds)
+        branch_factor = model_output.action_probs.shape[1]
+        if self._expand_all_children:
+            tree_size = (self._num_simulations + 1) * branch_factor + 1
+        elif self._expand_all_root_children:
+            tree_size = self._num_simulations + 1 + branch_factor
+        else:
+            tree_size = self._num_simulations + 1
+
+        trees = _MCTSTree(tree_size, model_output, self._known_value_bounds)
```

Comment:
Not related to this PR, but suggest renaming to `_MCTSTrees` or `_MCTSTreeBatch` will be slightly more friendly to the new reader.